src/Brush.elm

```diff
@@ -0,0 +1,1102 @@
+                        (D.map4
+                            (\pos meta alt shift ->
+                                let
+                                    el =
+                                        if keysEnabled && meta then
+                                            OverlayElement
+
+                                        else
+                                            element
+
+                                    mode =
+                                        if el == SelectionElement then
+                                            Drag
+
+                                        else if keysEnabled && alt then
+                                            Center
+
+                                        else
+                                            Handle
+                                in
+                                { message = tagger (MouseDown mode (keysEnabled && shift) element pos)
+                                , stopPropagation = True
+                                , preventDefault = False
+                                }
+                            )
+                            Events.decodeMousePosition
+                            (D.field "metaKey" D.bool)
+                            (D.field "altKey" D.bool)
+                            (D.field "shiftKey" D.bool)
+                        )
+                    ]
+
+                Just _ ->
+                    []
+           )
+
+
+{-| Initializes a brush that allows brusing in the X axis.
+-}
+initX : Extent -> Brush OneDimensional
+initX =
+    init True False
+
+
+{-| Initializes a brush that allows brusing in the Y axis.
+-}
+initY : Extent -> Brush OneDimensional
+initY =
+    init False True
+
+
+{-| Initializes a two dimensional brush.
+-}
+initXY : Extent -> Brush TwoDimensional
+initXY =
+    init True True
+
+
+init : Bool -> Bool -> Extent -> Brush dimension
+init x y extent =
+    Brush { drag = Nothing, keysEnabled = True, shifting = False, x = x, y = y, selection = Nothing, extent = extent }
+
+
+{-| By default the brush will use the meta/alt and shift keys to change behavior. You can disable this with this function.
+-}
+keyboardModifiersEnabled : Bool -> Brush dimension -> Brush dimension
+keyboardModifiersEnabled enabled (Brush model) =
+    Brush { model | keysEnabled = enabled }
+
+
+{-| Exposes the selection for a single dimensional brush, where the first number should always be less than the second.
+-}
+selection1d : Brush OneDimensional -> Maybe ( Float, Float )
+selection1d (Brush { selection, x }) =
+    Maybe.map
+        (\{ top, left, bottom, right } ->
+            if x then
+                ( min right left, max right left )
+
+            else
+                ( min top bottom, max top bottom )
+        )
+        selection
+
+
+{-| Exposes the selection for a two dimensional brush.
+-}
+selection2d : Brush TwoDimensional -> Maybe Extent
+selection2d (Brush { selection }) =
+    selection
+
+
+{-| -}
```

Comment:
Do we need a description here?

Reply:
What would you suggest? At the moment it's a bit of a pointless value, but I'm hoping to add a `animated` function in a later iteration (see how this works in the Zoom module).